commit 8a8dd919e116cce106891ada5279ff4e1ca59bd6
Author: Jan Safranek <jsafrane@redhat.com>
Date:   Mon Jan 27 13:01:51 2014 +0100

    Remove volume group metadata on DeleteStoragePool/DeleteVG.
    
    DeleteMDRAID does the same and we want to have the API consistent.
    
    In addition, add some checks around printing of 'action.device', some devices
    cannot be printed, e.g. those who are being deleted.

diff --git a/src/lmi/storage/LMI_StorageConfigurationService.py b/src/lmi/storage/LMI_StorageConfigurationService.py
index 13aa06b..c17ff67 100644
--- a/src/lmi/storage/LMI_StorageConfigurationService.py
+++ b/src/lmi/storage/LMI_StorageConfigurationService.py
@@ -1287,8 +1287,12 @@ class LMI_StorageConfigurationService(ServiceProvider):
         storage.assert_unused(self.storage, [pool])
 
         # finally delete it
-        action = blivet.ActionDestroyDevice(pool)
-        storage.do_storage_action(self.storage, [action])
+        actions = []
+        actions.append(blivet.ActionDestroyDevice(pool))
+        # Destroy also all formats on member devices.
+        for parent in pool.parents:
+            actions.append(blivet.deviceaction.ActionDestroyFormat(parent))
+        storage.do_storage_action(self.storage, actions)
 
         ret = self.Values.DeleteVG.Job_Completed_with_No_Error
         job.finish_method(
diff --git a/src/lmi/storage/util/storage.py b/src/lmi/storage/util/storage.py
index 02413d7..1898c5b 100644
--- a/src/lmi/storage/util/storage.py
+++ b/src/lmi/storage/util/storage.py
@@ -129,8 +129,14 @@ def do_storage_action(storage, actions):
     need_reset = False
 
     for action in actions:
-        LOG().trace_info("Running action " + str(action))
-        LOG().trace_info("    on device " + repr(action.device))
+        LOG().trace_info("Running action %s", str(action))
+        try:
+            LOG().trace_info("    on device %s", repr(action.device))
+        except:
+            # ignore printing errors
+            LOG().trace_info("    on device %s [cannot print full device info]",
+                    repr(action.device.path))
+            pass
 
         if (isinstance(action.device, blivet.devices.PartitionDevice)
                 and isinstance(action,
